src/relax/ir/block_builder.cc

```diff
@@ -484,6 +506,67 @@ class BlockBuilderNode::ExprNormalizer : public ExprFunctor<Expr(const Expr&)> {
     return Type();
   }
 
+  // Helper function to check if the provided shape and type is consistent.
+  // Throw internal exceptions if they are not consistent.
+  void CheckShapeTypeConsistency(const Optional<ObjectRef>& opt_shape, const Type& type) {
+    if (!type.defined() || type->IsInstance<ShapeTypeNode>() || type->IsInstance<FuncTypeNode>() ||
```

Comment:
Even though it's checked on line 548 that VoidType should have 0-length Tuple as its shape, does it make more sense if VoidType does not have shape, and add it to this list?

Reply:
Sorry I didn’t get your point here. We know that VoidType is represented as a TupleType with empty fields, but when given a TupleType of empty fields, we don’t know if it should be understood as VoidType or a real TupleType. So in line 548 we check if the shape is undefined to differentiate the type - whether it is a VoidType or not. So would you like ask the question again?